src/transaction/log_replication.cpp

```diff
@@ -271,6 +276,53 @@ namespace cublog
       }
   }
 
+  void replicator::wait_past_target_lsa (const log_lsa &a_target_lsa)
+  {
+    if (m_parallel_replication_redo == nullptr)
+      {
+	// sync, check up-front
+	{
+	  std::lock_guard<std::mutex> lockg { m_redo_lsa_mutex };
+	  if (m_redo_lsa > a_target_lsa)

```

Comment:
`mutex` is needed for `condition_variable` waiting. Making `m_redo_lsa` atomic would have no effect.
It will be possible to make `m_redo_lsa` atomic and drop the mutex with changes for http://jira.cubrid.org/browse/LETS-105 (I added a comment).